ocaml-lsp-server/test/e2e-new/documentation.ml

```diff
@@ -0,0 +1,100 @@
+                 identifier" =
+  let source = "type tree (** This is a comment *)" in
+  let line = 0 in
+  let character = 7 in
+  Util.test ~line ~character source;
+  [%expect
+    {| { "doc": { "kind": "plaintext", "value": "This is a comment" } } |}]
+
+let%expect_test "Documentation of simple type with contentFormat set to \
+                 markdown" =
+  let source = "type tree (** This is another comment *)" in
+  let line = 0 in
+  let character = 7 in
+  let contentFormat = "markdown" in
+  Util.test ~line ~character ~contentFormat source;
+  [%expect
+    {| { "doc": { "kind": "markdown", "value": "This is another comment" } } |}]
+
+let%expect_test "Documentation of simple type with an identifier and \
+                 contentFormat" =
+  let source =
+    "{|type tree (** This is another comment *)\n\
+    \  List.iter ~f:(fun x -> x*x) [2;4]|}"
+  in
+  let line = 0 in
+  let character = 7 in
+  let identifier = "List" in
+  let contentFormat = "markdown" in
+  Util.test ~line ~character ~identifier ~contentFormat source;
+  [%expect
+    {|
+    {
+      "doc": {
+        "kind": "markdown",
+        "value": "List operations.\n\n   Some functions are flagged as not tail-recursive.  A tail-recursive\n   function uses constant stack space, while a non-tail-recursive function\n   uses stack space proportional to the length of its list argument, which\n   can be a problem with very long lists.  When the function takes several\n   list arguments, an approximate formula giving stack usage (in some\n   unspecified constant unit) is shown in parentheses.\n\n   The above considerations can usually be ignored if your lists are not\n   longer than about 10000 elements.\n\n   The labeled version of this module can be used as described in the\n   {!StdLabels} module."
+      }
+    } |}]
+
+let%expect_test "Documentation of simple type with an identifier and no \
+                 contentFormat" =
+  let source =
+    "{|type tree (** This is another comment *)\n\
+    \  List.iter ~f:(fun x -> x*x) [2;4]|}"
+  in
+  let line = 0 in
+  let character = 7 in
+  let identifier = "List" in
+  Util.test ~line ~character ~identifier source;
+  [%expect
+    {|
+    {
+      "doc": {
+        "kind": "plaintext",
+        "value": "List operations.\n\n   Some functions are flagged as not tail-recursive.  A tail-recursive\n   function uses constant stack space, while a non-tail-recursive function\n   uses stack space proportional to the length of its list argument, which\n   can be a problem with very long lists.  When the function takes several\n   list arguments, an approximate formula giving stack usage (in some\n   unspecified constant unit) is shown in parentheses.\n\n   The above considerations can usually be ignored if your lists are not\n   longer than about 10000 elements.\n\n   The labeled version of this module can be used as described in the\n   {!StdLabels} module."
+      }
+    } |}]
```

Comment:
Can you add a test where you Shadow `List` for your own implementation (and documentation)?

Reply:
@xvw When I shadow `List` and run this doc query, for some reason it returns the docs for original List module. Could this be a scope issue?

Reply:
With, for example, the `Base`'s one ?

Reply:
I updated the test. I added a todo comment to investigate on the Merlin side.